### PR TITLE
RD-3442 Moved user menu from top to side bar

### PR DESCRIPTION
**Pull request by kubama**
## Description
Moved user menu from top to side bar

## Screenshots / Videos
image

## Checklist

- [X] My code follows the UI Code Style.
- [X] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1566/ - I fixed the failing component test and retested it locally, I'm not planning another full system tests execution

 

## Documentation
Will be covered in separate ticket

### CODE REVIEW

File: app/components/sidebar/UserMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    return (
+        <>
+            <SideBarItem
+                icon={
+                    <Label
+                        style={{ float: 'none', marginLeft: -10, marginTop: -5, marginRight: 9, width: '1.2em' }}
+                        circular
+                    >
+                        {username.substr(0, 2)}
```
Comment (qooban) [inline app/components/sidebar/UserMenu.tsx:84]: According to the mockups, I think we should uppercase it.

**Review by qooban — COMMENTED** [review]
Tested UI - looks good 👍🏻

Added some comments.

File: test/cypress/integration/templateManagement/pages_edit_spec.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@ describe('Page management', () => {
     it('allows admin users to create and modify pages', () => {
         const pageName = 'Page 1';
 
-        cy.get('.usersMenu').click();
-        cy.get('.usersMenu').contains('Template Management').click();
+        cy.contains('admin').click({ force: true });
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/templateManagement/pages_edit_spec.ts]: Added `clickSidebarItem` command

File: test/cypress/integration/templateManagement/pages_edit_spec.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@ describe('Page management', () => {
...
+        cy.contains('Template Management').click();
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/templateManagement/pages_edit_spec.ts]: Added `goToTemplateManagement` command

File: app/components/sidebar/UserMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                    </Label>
+                }
+                label={username}
+                onClick={toggleExpand}
+                expandable
+                expanded={expanded}
+            />
+
+            {expanded && (
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/sidebar/UserMenu.tsx:94]: Items rendered as children are actually siblings in terms of component tree

File: app/components/shared/PasswordModal.tsx [hunk 5]
```diff
@@ -1,18 +1,26 @@
 // @ts-nocheck File not migrated fully to TS
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/shared/PasswordModal.tsx:1]: It's not fully migrated yet

File: app/actions/managers.ts [hunk 6]
```diff
@@ -121,7 +121,7 @@ function doLogout(err) {
     };
 }
 
-export function logout(err, path) {
+export function logout(err?, path?) {
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/actions/managers.ts:124]: In the file that I migrated `logout` function is called  with no arguments, so I had to make the arguments optional to satisfy TS compiler

File: app/components/sidebar/SideBarItemIcon.tsx [hunk 7]
```diff
@@ -7,9 +7,11 @@ import { Icon } from '../basic';
 const SideBarItemIcon = styled(({ name, ...rest }: IconProps) => <Icon name={name ?? 'expand'} {...rest} />)`
     &&&&&& {
         margin-left: -10px;
-        margin-right: 4px;
+        margin-right: 9px;
+        margin-top: -2px;
         float: none;
         width: 1.2em;
+        font-size: 19px;
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/sidebar/SideBarItemIcon.tsx:14]: After adding the 'user initials' icon the other icons just looked too small, so I decided to enlarge them at this point

File: test/cypress/integration/cluster_status_spec.ts [hunk 8]
```diff
@@ -58,9 +58,9 @@ describe('Cluster Status', () => {
 
             cy.log('Check system status icon');
             if (expectedStatusIndicatorColor) {
-                cy.contains('Health').find('.circle').should('have.class', expectedStatusIndicatorColor);
+                cy.contains('.item', 'Health').find('.circle').should('have.class', expectedStatusIndicatorColor);
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/cluster_status_spec.ts:61]: Actually the 'item' element is the one I specifically wanted to refer here - it's the deepest element containing 'Health' text and circle icon at the same time.

File: app/components/sidebar/UserMenu.tsx (hunk 1 above)
Comment (kubama) [reply app/components/sidebar/UserMenu.tsx]: Done

File: app/components/sidebar/UserMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    return (
+        <>
+            <SideBarItem
+                icon={
+                    <Label
+                        style={{ float: 'none', marginLeft: -10, marginTop: -5, marginRight: 9, width: '1.2em' }}
+                        circular
+                    >
+                        {username.substr(0, 2)}
+                    </Label>
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/sidebar/UserMenu.tsx]: Done

File: app/reducers/managerReducer.ts [hunk 10]
```diff
@@ -13,12 +13,14 @@ export interface ManagerData {
     };
     clusterStatus: Record<string, any>;
     err: any;
+    isLdap: boolean;
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/reducers/managerReducer.ts:16]: Done

File: app/components/sidebar/SideBarDropdownItem.tsx [hunk 11]
```diff
@@ -12,18 +12,7 @@ interface SideBarDropdownItemProps extends DropdownProps {
...
-                    {icon}
-                    {label}
-                </SideBarItem>
-            }
-            pointing="left"
-            icon={null}
-            fluid
-            onOpen={onOpen}
-        >
+        <Dropdown trigger={<SideBarItem icon={icon} label={label} />} pointing="left" icon={null} fluid onOpen={onOpen}>
```
Comment (NorbertAmartus) [inline app/components/sidebar/SideBarDropdownItem.tsx:15]: As I can see from the `DropdownProps`, the icon typing looks like this:
```
  /** Shorthand for Icon. */
  icon?: any
```

So we should be able to not explicitly declare `icon` as `null`

File: app/components/sidebar/UserMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,132 @@
+import _ from 'lodash';
+import React, { FunctionComponent } from 'react';
```
Comment (NorbertAmartus) [inline app/components/sidebar/UserMenu.tsx:2]:
```suggestion
import React from 'react';
import type { FunctionComponent } from 'react';
```

File: app/components/sidebar/SideBarDropdownItem.tsx (hunk 11 above)
Comment (kubama) [reply app/components/sidebar/SideBarDropdownItem.tsx:15]: When `icon` prop is not specified then default icon is used, passing `null` hides the icon

Comment (NorbertAmartus) [reply app/components/sidebar/SideBarDropdownItem.tsx:15]: Okay, thanks for the explanation!

**Review by NorbertAmartus — APPROVED** [review]

File: app/components/sidebar/UserMenu.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/sidebar/UserMenu.tsx:88]: Right, they are siblings, I didn't spot it here. I got confused with `children` passed to `SideBarItem` as they are rendered the following way:
```
        <SideBarItemWrapper>
            {/* eslint-disable-next-line react/jsx-props-no-spreading */}
            <Menu.Item style={menuItemStyle} {...rest}>
                {typeof icon === 'string' ? <SideBarItemIcon name={icon as SemanticICONS} /> : icon}
                {label && <span style={{ verticalAlign: 'top' }}>{label}</span>}
                {children}
                {expandable && (
                    <Icon
                        name="dropdown"
                        rotated={expanded ? undefined : 'counterclockwise'}
                        style={{ position: 'absolute', right: 12, margin: 0 }}
                    />
                )}
            </Menu.Item>
        </SideBarItemWrapper>
```
Do we need them, then? Maybe that `{expanded && ( ... ) }` part could be moved somehow inside `SideBarItem`?

Of course it's not something for this PR, especially that others may be waiting for system tests fix. But let's discuss.

**Review by qooban — APPROVED** [review]
Thanks for addressing comments 👍🏻  
I have left one discussion open. It's not blocking this PR from being merged.

**Comment by kubama** [comment]
> Do we need them, then? Maybe that {expanded && ( ... ) } part could be moved somehow inside SideBarItem?

The reason for such design is `dnd-kit` framework limitations. In order to implement moving pages to/from page groups all pages have to be siblings regardless of their actual nesting. As a result all sidebar items are siblings even if some of them are actually rendered as child items.

**Pull request merged**